pkg/systemlogmonitor/log_monitor.go

```diff
@@ -73,13 +73,35 @@ func NewLogMonitorOrDie(configPath string) types.Monitor {
 		glog.Fatalf("Failed to validate matching rules %+v: %v", l.config.Rules, err)
 	}
 	glog.Infof("Finish parsing log monitor config file: %+v", l.config)
+
 	l.watcher = logwatchers.GetLogWatcherOrDie(l.config.WatcherConfig)
 	l.buffer = NewLogBuffer(l.config.BufferSize)
 	// A 1000 size channel should be big enough.
 	l.output = make(chan *types.Status, 1000)
+
+	initializeProblemMetricsOrDie(l.config.Rules)
 	return l
 }
 
+// initializeProblemMetricsOrDie creates problem metrics for all problems and set the value to 0,
+// panic if error occurs.
+func initializeProblemMetricsOrDie(rules []systemlogtypes.Rule) {
+	for _, rule := range rules {
+		if rule.Type == types.Temp {
+			err := problemmetrics.GlobalProblemMetricsManager.NewTemporaryProblems(rule.Reason, 0)
+			if err != nil {
+				glog.Fatalf("Failed to initialize temporary problem metrics for %q: %v", rule.Reason, err)
+			}
+		} else {
+			err := problemmetrics.GlobalProblemMetricsManager.NewPermanentProblem(rule.Condition, rule.Reason, false)
```

Comment:
SG. I like it. Done.

Reply:
In that case, I will rename `NewTemporaryProblems()` and `NewPermanentProblem()` to `IncrementProblemCounter()` and `SetProblemGauge()`.
Because new we are now reporting to problem counter for permanent problems as well.